whisper.cpp

```diff
@@ -4589,7 +4589,7 @@ static void whisper_process_logits(
 
         // suppress sot and nosp tokens
         logits[vocab.token_sot]  = -INFINITY;
-        logits[vocab.token_nosp] = -INFINITY; // TODO: ignore this token for now
+        // logits[vocab.token_nosp] = -INFINITY; // Uncommenting this would produce hallucinations on silent audio
```

Comment:
Although it is said that `token_nosp` is the direction to solve the hallucination, it is definitely problematic for you to cancel the suppress `token_nosp`. First of all, we only hope that the output of the model contains meaningful and visible tokens (except for timestamps). Your cancellation of suppress `token_nosp` will cause this token to possibly appear in the output of the model, which is something we do not want to see. Secondly, the key to solving hallucination lies in finding a way to skip silence. `token_nosp` is used to tell you how likely it is that this segment is silent, so that we can skip silence. Therefore, merely cancelling suppress `token_nosp` without any other action cannot solve hallucination.